### DISCUSSION TITLE
math: curly brackets $\{...\}$ not working

### DISCUSSION THREAD

**Discussion by nschloe** [community/community#16993] [Other Feature Feedback, Questions, & Ideas] [answered]
```
$$
\{a\}
$$
```
##### Test

- Inline: $\{a\}$

- Display with dollar-dollar:

  $$
  \{a\}
  $$

- Display with triple-backtick:

  ```math
  \{a\}
  ```

Screenshot:

![screenshot](https://user-images.githubusercontent.com/181628/175816983-1868418a-a846-4e1d-895b-23527889910b.png)

---

_Edit:_ I guess I know why this happens:

The Markdown renderer gets called first. It doesn't know `$`-syntax, so it "sanitzes" the math content like any other content. This means removing `\` in front of everything that's not a letter. There go your `\{`, `\:`, `\;` etc. in math mode.

**Comment by Simran-B** [comment]
I think you are right about the input getting sanitized, removing unknown backslash escapes. If you escape the backslashes themselves, then it works:

```
$$
\\{a\\}
$$
```

$$
\\{a\\}
$$

**Reply by nschloe** [reply]
Sure, you can replace each backslash by two. That's not LaTeX though. Also, you don't need it if a backslash is followed by a letter, e.g.,
```
$\leq$
```
$$
\leq
$$

**Comment by pedritomelenas** [comment]
I solved this problem using `\lbrace` and `\rbrace`: $\lbrace a \rbrace$

$$
\lbrace a \rbrace
$$

Hope this helps.

**Reply by EtorArza** [reply]
MVP of the week

**Comment by nschloe** [comment, marked as answer]
This is fixed with the new backtick math syntax:

````markdown
$`\{a\}`$

```math
\{a\}
```
````

> $`\{a\}`$
>
> ```math
> \{a\}
> ```

**Reply by samestep** [reply]
Thank you! It would be nice if the same also worked in VS Code, but unfortunately it doesn't: microsoft/vscode#208430

**Reply by samm82** [reply]
Note that the backticks should be adjacent to the `$` and not the braces themselves, such as in expressions like $`a \in \{a, b, c\}`$ (hopefully that makes sense; it confused me initially).
